src/tests/ftest/tags.py

```diff
@@ -267,6 +285,102 @@ def __parse_avocado_tags(text):
         return set(','.join(tag_strings).split(','))
 
 
+class TestConfig():
+    """Represent the test configuration defined in the test yaml file."""
+
+    def __init__(self, path, test_name, verbose=False):
+        """Initialize the tag mapping.
+
+        Args:
+            path (str): test file path or config path
+            test_name (str): test method name to get timeout for
+        """
+        self.__data = {}
+        with open(path.replace('.py', '.yaml'), 'r') as f:
+            self.__data = yaml.load(f.read(), Loader=AvocadoYamlLoader)
+        self.__test_name = test_name
+        self.__verbose = verbose
+
+    def __display(self, message):
+        """Display the message if verbose is turned on.
+
+        Args:
+            message (str): message to display
+        """
+        if self.__verbose:
+            print(message)
+
+    def __filter_value(self, key_path, key_types=None):
+        """Get the test yaml data value for a given key.
+
+        Args:
+            key_path (list): path names from which to match the key where the last entry is the
+                key to lookup in the test yaml data.
+            key_types (list, optional): key object types to match. Defaults to None.
+
+        Returns:
+            list: values found for the key
+        """
+        self.__display(f"__filter_value: key_path={key_path}, key_types={key_types}")
```

Comment:
I think it's better to not leave this debug here. Especially since it requires a helper method + new argument in init. This is only ever needed if there is a bug. And to guard against bugs its better to have unit tests than debug printouts